plugins/polio/models.py

```diff
@@ -961,3 +967,268 @@ class VaccineAuthorization(SoftDeletableModel):
+                "import_raw_data": row.to_dict(),
+                "import_source": self,
+                "lineage": importer.clean_str(row["LINEAGE"]),
+                "org_unit": org_unit,
+                "site_name": importer.clean_str(row["SITE_NAME/GEOCODE"]),
+                "source": importer.clean_source(row["SOURCE(AFP/ENV/CONTACT/HC)"]),
+                "vdpv_category": importer.clean_vdpv_category(row["VDPV_CATEGORY"]),
+                "vdpv_nucleotide_diff_sabin2": importer.clean_str(row["VDPV_NUCLEOTIDE_DIFF_SABIN2"]),
+            }
+            _, created = Notification.objects.get_or_create(
+                epid_number=importer.clean_str(row["EPID_NUMBER"]),
+                defaults=kwargs,
+            )
+            if not created:
+                errors.append(row.to_dict())
+
+        self.status = self.Status.DONE
+        self.errors = errors
+        self.updated_at = timezone.now()
+        self.save()
+
+
+@task_decorator(task_name="create_polio_notifications_async")
+def create_polio_notifications_async(pk: int, task: Task = None) -> None:
+    task.report_progress_and_stop_if_killed(progress_message="Importing polio notifications…")
+    user = task.launcher
+    notification_import = NotificationImport.objects.get(pk=pk)
+    notification_import.create_notifications(created_by=user)
+    num_created = Notification.objects.filter(import_source=notification_import).count()
+    task.report_success(message=f"{num_created} polio notifications created.")
+
+
+class NotificationXlsxImporter:
+    def __init__(self):
+        self.countries_cache = None
+        self.regions_cache = None
+        self.districts_cache = None
+
+    def build_org_unit_caches(self) -> Tuple[defaultdict, defaultdict, defaultdict]:
+        from plugins.polio.api.common import make_orgunits_cache
+
+        districts = OrgUnit.objects.filter(
+            org_unit_type__category="DISTRICT", validation_status=OrgUnit.VALIDATION_VALID
+        ).select_related("org_unit_type")
+
+        regions = OrgUnit.objects.filter(
+            OrgUnit.objects.parents_q(districts),
+            org_unit_type__category="REGION",
+            validation_status=OrgUnit.VALIDATION_VALID,
+            path__depth=2,
+        ).select_related("org_unit_type")
+
+        countries = OrgUnit.objects.filter(
+            OrgUnit.objects.parents_q(districts),
+            org_unit_type__category="COUNTRY",
+            validation_status=OrgUnit.VALIDATION_VALID,
+            path__depth=1,
+        ).select_related("org_unit_type")
+
+        districts_cache = make_orgunits_cache(districts)
+        regions_cache = make_orgunits_cache(regions)
+        countries_cache = make_orgunits_cache(countries)
+
+        return countries_cache, regions_cache, districts_cache
+
+    def find_org_unit_in_caches(self, country_name: str, region_name: str, district_name: str) -> Union[None, OrgUnit]:
+        from plugins.polio.api.common import find_orgunit_in_cache
```

Comment:
Why is the import local? (I guess something with import orders, but not sure)

Reply:
The purpose is to avoid a circular import.